src/components/NewCommentForm.tsx

```diff
@@ -47,25 +46,31 @@ export const NewCommentForm: React.FC<Props> = ({ onSubmit }) => {
   const handleSubmit = async (event: React.FormEvent) => {
     event.preventDefault();
 
-    setErrors({
-      name: !name,
-      email: !email,
-      body: !body,
-    });
+    if (selectedPost) {
```

Comment:
You should consider handling the case when `selectedPost` is `null` or `undefined` at the start of the function to avoid unnecessary computations and improve code readability. You can return early if `selectedPost` is not defined.